packages/server/src/app.ts

```diff
@@ -14,21 +14,27 @@ import Koa from "koa"
 import { Server } from "http"
 import { startup } from "./startup"
 
-let app: Koa, server: Server
+let app: Koa, server: Server, started: Promise<void>
 
 async function start() {
   const koa = createKoaApp()
-  app = koa.app
-  server = koa.server
-  // startup includes automation runner - if enabled
-  await startup({ app, server })
+  if (!started) {
+    started = new Promise((resolve, reject) => {
+      app = koa.app
+      server = koa.server
+      // startup includes automation runner - if enabled
+      startup({ app, server }).then(resolve).catch(reject)
+    })
+  }
+  await started
 }
 
 start().catch(err => {
   console.error(`Failed server startup - ${err.message}`)
   throw err
 })
 
-export function getServer(): Server {
+export async function getServer(): Promise<Server> {
```

Comment:
This is a key change. In order for tests to wait for everything to properly initialise, this had to be made async so it can be `await`ed.